causallib/time_varying/GFormula.py

```diff
@@ -172,23 +172,17 @@ def estimate_population_outcome(self,
                                     ) -> pd.DataFrame:
 
         """
-            Calculates population outcome for each subgroup stratified by treatment assignment. #TODO confirm
+            Calculates population outcome for each subgroup stratified by treatment assignment.
+            Backlog: Support for multiple treatment strategies. Like a list of  "always_treated"  and "never_treated".
         """
-        raise NotImplementedError()
-
-        unique_treatment_values = a.unique()
-        res = {}
-        for treatment_value in unique_treatment_values:
-            assignment = pd.Series(data=treatment_value, index=X.index, name=a.name)
-            individual_prediction_curves = self.estimate_individual_outcome(X=X,
-                                                                            a=assignment,
-                                                                            t=t,
-                                                                            y=y,
-                                                                            treatment_strategy=treatment_strategy,
-                                                                            timeline_start=timeline_start,
-                                                                            timeline_end=timeline_end)
-            res[treatment_value] = individual_prediction_curves.mean(axis=0).squeeze(axis=0)  # returns n_steps * (X-dim * a-dim)
-
+        individual_prediction_curves = self.estimate_individual_outcome(X=X,
+                                                                        a=a,
+                                                                        t=t,
+                                                                        y=y,
+                                                                        treatment_strategy=treatment_strategy,
+                                                                        timeline_start=timeline_start,
+                                                                        timeline_end=timeline_end)
+        res = individual_prediction_curves.mean(axis=0).squeeze(axis=0)  # returns n_steps * (X-dim * a-dim)
```

Comment:
What is the return type of `individual_prediction_curves` --- is that a dataframe with one column for id's? if so you may need to groupby the id rather than mean over axis

Reply:
The return type of this method is a dataframe ( pat_id * n_steps *  covariates), where the result has already been aggregated per id's from estimate_individual_outcome() in line 98.

Reply:
@pchakraborty1 
The return type of `individual_outcome_estimation` has been changed. Now, the return DataFrame will be of shape `n_steps * n_cols`, with an additional column `pat_id`. Then, we can simple aggregate the result in `population_outcome_estimation` by grouping across `time` column. 

Please have the code reviewed.